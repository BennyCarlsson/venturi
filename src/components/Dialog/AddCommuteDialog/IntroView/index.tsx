import Button from 'components/Button'
import Heading from 'components/Heading'
import image from 'assets/images/bus-stop/background-bus-stop.jpg'
import {
  BottomPart,
  IntroViewWrapper,
  StyledImage,
  StyledParagraph,
  StyledTitleHeading,
} from './styles'

type IntroViewProps = {
  goToChooseDestionationView: () => void
}

const IntroView = ({ goToChooseDestionationView }: IntroViewProps) => {
  return (
    <IntroViewWrapper>
      <span>
        <StyledTitleHeading
          headingType={'h1'}
          fontSize={28}
          color='#719FB0'
          fontWeight={400}
        >
          Venturi
        </StyledTitleHeading>
        <Heading
          headingType='h2'
          fontSize={28}
          fontWeight={700}
          color={'#35F39'}
        >
          Commute trip planner
        </Heading>
        <StyledParagraph fontSize={14} fontWeight={500}>
          Have a quick way to get information about your most common trip.
        </StyledParagraph>
        <StyledParagraph fontSize={14} fontWeight={500}>
          The specified trip will be easily acceible with current status and
          more.
        </StyledParagraph>
      </span>
      <BottomPart>
        <StyledImage src={image} />
        <Button onClick={goToChooseDestionationView}>Get Started!</Button>
      </BottomPart>
    </IntroViewWrapper>
  )
}

export default IntroView
