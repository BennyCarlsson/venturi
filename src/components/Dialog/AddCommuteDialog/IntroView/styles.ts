import styled from 'styled-components'
import Heading from 'components/Heading'
import Paragraph from 'components/Paragraph'

export const IntroViewWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 8px;
`
export const StyledParagraph = styled(Paragraph)`
  margin-top: 24px;
`

export const StyledTitleHeading = styled(Heading)`
  margin-bottom: 8px;
`

export const BottomPart = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const StyledImage = styled.img`
  margin-top: 48px;
  margin-bottom: 64px;
  width: 100%;
  max-width: 800px;
`
