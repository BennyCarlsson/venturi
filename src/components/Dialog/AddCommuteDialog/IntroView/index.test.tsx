import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import IntroView from '.'

describe('<IntroView />', () => {
  const mockGoToChooseDestinationView = jest.fn()
  it('should match snapshot', () => {
    const { asFragment } = render(
      <IntroView goToChooseDestionationView={mockGoToChooseDestinationView} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call goToChooseDestionationView onClick', async () => {
    render(
      <IntroView goToChooseDestionationView={mockGoToChooseDestinationView} />
    )
    expect(mockGoToChooseDestinationView).not.toHaveBeenCalledTimes(1)
    await userEvent.click(screen.getByRole('button'))
    expect(mockGoToChooseDestinationView).toHaveBeenCalledTimes(1)
  })
})
