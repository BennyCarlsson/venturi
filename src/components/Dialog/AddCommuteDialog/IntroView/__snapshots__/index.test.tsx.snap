// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroView /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw jITuUJ"
  >
    <span>
      <h1
        class="sc-gsTCUz crbHUW sc-jrAGrp ciFszB"
        color="#719FB0"
        font-size="28"
        font-weight="400"
      >
        Venturi
      </h1>
      <h2
        class="sc-dlfnbm iYlpBT"
        color="#35F39"
        font-size="28"
        font-weight="700"
      >
        Commute trip planner
      </h2>
      <p
        class="sc-iBPRYJ jNAkW sc-pFZIQ dNSvNb"
        font-size="14"
        font-weight="500"
      >
        Have a quick way to get information about your most common trip.
      </p>
      <p
        class="sc-iBPRYJ jNAkW sc-pFZIQ dNSvNb"
        font-size="14"
        font-weight="500"
      >
        The specified trip will be easily acceible with current status and more.
      </p>
    </span>
    <div
      class="sc-kEjbxe bpoeqT"
    >
      <img
        class="sc-iqHYGH iGFhbH"
        src="background-bus-stop.jpg"
      />
      <button
        class="sc-bdfBwQ ssyvv"
      >
        Get Started!
      </button>
    </div>
  </div>
</DocumentFragment>
`;
